gutils/gimagereadras.c

```diff
@@ -328,13 +321,25 @@ GImage *GImageReadRas(char *filename) {
 	}
     if ( ret!=NULL ) {
 	    /* All okay if reached here, return converted image */
-	    fclose(fp);
 	    return( ret );
     }
 
 errorGImageReadRas:
-    fprintf(stderr,"Bad input file \"%s\"\n",filename );
+    fprintf(stderr,"Bad input file\n");
     GImageDestroy(ret);
-    fclose(fp);
     return( NULL );
 }
+
```

Comment:
You don't need all of these, and I'd prefer to see them refactored a bit:

1. Make one function "GImageRead_Wrapper(char * filename, GImageReadFunc *func) ...
2. Typedef `GImageReadFunc` as something like `GImage *(*GImageReadFunc)(File *, int *);`, where the second argument (when not NULL) returns non-zero on error and 0 on success.
3. Modify your _ functions to take the second argument. If it is null, print the error, If it is non-null, set the success condition and return without printing.
4. Modify the (now single) wrapper to call the passed function pointer with a non-null second argument (e.g. `int success;` and then `&success` as the argument. Check for success, and on error print the error message with the filename.
5. Modify the calls to the _ functions in GImageReadBuf to pass NULL as the second argument. 

This way there is less code and you don't loose the filename in the error message when it is available.